### PR TITLE
Fix an issue where summary keys were not consistently named

### PR DESCRIPTION
**Pull request by eivindjahren**
Ensure consistent naming of summary keys. This also fixes an issue
where history keys were not handled correctly: e.g.

the history key of "BOPR:1,1,3" was before interpreted to be
"BOPR:1,1,3H" but is now interpreted as "BOPRH:1,1,3".

Whether that always makes sense in all simulators is not confirmed, but
it "BOPR:1,1,3H" is guaranteed to not be found in the summary file. The
keys that did work correctly before, `FIELD`, `OTHER`, `GROUP` and `WELL`
still works the same way.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/ert_config.py [hunk 1]
```diff
@@ -651,12 +651,10 @@ def preferred_num_cpu(self) -> int:
...
     def _create_observations(self) -> EnkfObs:
         obs_config_file = self.model_config.obs_config_file
-        obs_time_list: List[datetime] = []
+        obs_time_list: Sequence[datetime] = []
         if self.ensemble_config.refcase is not None:
             refcase = self.ensemble_config.refcase
-            obs_time_list = [refcase.get_start_time()] + [
-                CTime(t).datetime() for t in refcase.alloc_time_vector(True)
-            ]
+            obs_time_list = [refcase[0]] + list(refcase[2])
```
Comment (andreas-el) [inline src/ert/config/ert_config.py:657]: This is so because of startdate, timemap ?

File: src/ert/config/ensemble_config.py [hunk 2]
```diff
@@ -150,28 +131,32 @@ def make_field(field_list: List[str]) -> Field:
...
-        summary_keys = [item for sublist in summary_list for item in sublist]
+        to_add = list(refcase_keys)
         for key in summary_keys:
-            if "*" in key and refcase:
-                optional_keys.extend(list(refcase.keys(pattern=key)))
+            if "*" in key:
+                for i, rkey in list(enumerate(to_add))[::-1]:
+                    if fnmatch(rkey, key) and rkey != "TIME":
+                        optional_keys.append(rkey)
+                        del to_add[i]
```
Comment (andreas-el) [inline src/ert/config/ensemble_config.py:149]: Ok since we are reversing iteration

**Review by andreas-el — CHANGES_REQUESTED** [review]
This is definitely moving in the right direction, but there are some failing tests that needs to be resolved at a minimum.

File: src/ert/config/ensemble_config.py (hunk 2 above)
Comment (eivindjahren) [reply src/ert/config/ensemble_config.py:149]: Yup.

**Review by andreas-el — APPROVED** [review]
Great! 🚀

**Pull request merged**